src/data-types/int.ts

```diff
@@ -15,28 +15,35 @@ const Int: DataType = {
     buffer.writeUInt8(4);
   },
 
-  writeParameterData: function(buffer, parameter, options, cb) {
+  writeParameterData: function(buffer, parameter, _options, cb) {
     if (parameter.value != null) {
       buffer.writeUInt8(4);
-      buffer.writeInt32LE(parseInt(parameter.value));
+      buffer.writeInt32LE(Number(parameter.value));
     } else {
       buffer.writeUInt8(0);
     }
+
     cb();
   },
 
   validate: function(value): number | null | TypeError {
     if (value == null) {
       return null;
     }
-    value = parseInt(value);
+
+    if (typeof value !== 'number') {
+      value = Number(value);
+    }
+
     if (isNaN(value)) {
       return new TypeError('Invalid number.');
     }
+
     if (value < -2147483648 || value > 2147483647) {
-      return new TypeError('Value must be between -2147483648 and 2147483647.');
+      return new TypeError('Value must be between -2147483648 and 2147483647, inclusive.');
     }
-    return value;
+
+    return value | 0;
```

Comment:
For `smallint`, `tinyint` and `int`, we use `| 0` here to force the value to be converted internally to a 32 bit integer in the V8 engine. I haven't benchmarked this, but it _might_ improve performance by giving V8 better optimization hints. 🤷‍♂

Reply:
We can't use the same thing for `bigint`, because `| 0` only works with the first 32 bits of a value and thus would lead to incorrect results.